.github/workflows/haskell.yml

```diff
@@ -5,38 +5,37 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v1
-    - uses: cachix/install-nix-action@v7
-    - uses: cachix/cachix-action@v4
+    - uses: mstksg/setup-stack@v1
+    - uses: actions/cache@v1
       with:
-        name: 47deg
-        signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
-
+        path: ~/.stack
+        key: dot-stack-${{ hashFiles('stack.yaml') }}
+        restore-keys: |
+          dot-stack-
+    - name: Get rdkafka
+      run: sudo apt-get install librdkafka-dev
+    - name: Compile with Stack
+      run: stack build
   test_stack_templates:
     runs-on: ubuntu-latest
     steps:
-      - uses: mstksg/setup-stack@v1
-
-      - uses: actions/checkout@v1
-
-      - uses: actions/cache@v1
-        with:
-          path: ~/.stack
-          key: dot-stack-${{ hashFiles('templates/*.hsfiles') }}
-          restore-keys: |
-            dot-stack-
-
-      - name: Generate Avro RPC server from Stack template
-        run: stack new muhaskellavro https://raw.githubusercontent.com/higherkindness/mu-haskell/master/templates/grpc-server-avro.hsfiles -p "author-email:haskell.curry@47deg.com" -p "author-name:Haskell Curry"
-        working-directory: /tmp
-
-      - name: Build the generated Avro RPC server
-        run: stack build
-        working-directory: /tmp/muhaskellavro
-
-      - name: Generate Protobuf RPC server from Stack template
-        run: stack new muhaskellprotobuf https://raw.githubusercontent.com/higherkindness/mu-haskell/master/templates/grpc-server-protobuf.hsfiles -p "author-email:haskell.curry@47deg.com" -p "author-name:Haskell Curry"
-        working-directory: /tmp
-
-      - name: Build the generated Protobuf RPC server
-        run: stack build
-        working-directory: /tmp/muhaskellprotobuf
+    - uses: actions/checkout@v1
+    - uses: mstksg/setup-stack@v1
+    - uses: actions/cache@v1
+      with:
+        path: ~/.stack
+        key: dot-stack-${{ hashFiles('templates/*.hsfiles') }}
```

Comment:
I think the caches for both jobs will be shared, because they are in the same workflow. So you might want to give them different names so they don't clash with each other?

If you give them a common prefix, e.g. `dot-stack-build-...` and `dot-stack-templates-...`, you can use that prefix as a fallback in `restore-keys`, so you should never end up with a completely cold cache:

```
restore-keys: |
  dot-stack-templates-
  dot-stack-
```

The action will try each prefix in turn until it finds a cache it can use. More details [here](https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key).